opentelemetry-sdk/tests/trace/test_trace.py

```diff
@@ -393,6 +404,38 @@ def test_attributes(self):
             self.assertEqual(root.attributes["attr-key2"], "val2")
             self.assertEqual(root.attributes["attr-in-both"], "span-attr")
 
+    def test_invalid_attribute_values(self):
+        with self.tracer.start_as_current_span("root") as root:
+            root.set_attribute("non-primitive-data-type", dict())
+            root.set_attribute(
+                "list-of-mixed-data-types-numeric-first",
+                [123, False, "string"],
+            )
+            root.set_attribute(
+                "list-of-mixed-data-types-non-numeric-first",
+                [False, 123, "string"],
+            )
+            root.set_attribute(
+                "list-with-non-primitive-data-type", [dict(), 123]
+            )
+
+            self.assertEqual(len(root.attributes), 0)
+
+    def test_check_sequence_helper(self):
+        # pylint: disable=protected-access
+        self.assertEqual(
+            trace.Span._check_sequence([1, 2, 3.4, "ss", 4]), "different type"
+        )
+        self.assertEqual(
+            trace.Span._check_sequence([1, 2, 3.4, dict(), 4]), "invalid type"
+        )
+        self.assertEqual(
+            trace.Span._check_sequence(["sw", "lf", 3.4, "ss"]),
+            "different type",
```

Comment:
Looking good. The returned values in my example were a suggestion only to show a solution for the inconsistency of the error logged and the usage of `Number`. Please feel free to change this helper method however it best suits your design approach.

Reply:
👍 These return values fit nicely in the log message so I'm fine leaving this as-is.